packages/Filters/src/components/Item/Item.js

```diff
@@ -0,0 +1,188 @@
+  } = props;
+  const { columns, data, filtersRef, onChangeOperator, operator, rulesByType } = React.useContext(FilterContext);
+  const I18n = useI18n();
+
+  const selectedColumnType = columns.find(({ id }) => id === selectedColumnId).type;
+
+  const selectOptions = React.useMemo(() => {
+    return selectedColumnType === types.columnTypes.SINGLE_SELECT
+      ? data.filter((obj, index, arr) => {
+          return arr.map(mapObj => mapObj[selectedColumnId]).indexOf(obj[selectedColumnId]) === index;
+        })
+      : null;
+  }, [selectedColumnType, data, selectedColumnId]);
+
+  function handleRemoveFilter() {
+    filtersRef.current.focus();
+    onDeleteFilter(id);
+  }
+
+  function handleChangeColumn(event) {
+    const newColumnId = event.target.value;
+    onChangeFilter(types.changeTypes.COLUMN, { id, columnId: newColumnId });
+  }
+
+  function handleChangeRule(event) {
+    onChangeFilter(types.changeTypes.RULE, { id, rule: event.target.value });
+  }
+
+  function handleChangeValue(e) {
+    const newInputtedValue = e.target.value;
+    onChangeFilter(types.changeTypes.FILTER_VALUE, { id, value: newInputtedValue });
+  }
+
+  function handleChangeDatePicker(momentDate) {
+    filtersRef.current.focus();
+
+    if (momentDate === null) {
+      handleChangeValue("");
+    } else {
+      handleChangeValue(momentDate.format(columns.find(({ id }) => id === selectedColumnId).momentParsingFormat));
+    }
+  }
+
+  function handleChangeBooleanFilterValue(event) {
+    onChangeFilter(types.changeTypes.FILTER_VALUE, { id, value: event.target.value === "true" });
+  }
+
+  function handleChangeSingleSelectFilterValue(event) {
+    onChangeFilter(types.changeTypes.FILTER_VALUE, { id, value: event.target.value });
+  }
+
+  function renderRuleField() {
+    switch (selectedColumnType) {
+      case types.columnTypes.BOOLEAN:
+        return I18n.t("filters.rules.is");
+      default:
+        return (
+          <sc.RuleSelect
+            as={Select}
+            onChange={handleChangeRule}
+            value={selectedRule}
+            data-pka-anchor="filters.ruleSelector"
+          >
+            {rulesByType[selectedColumnType].map(rule => (
+              <option key={rule} value={rule}>
+                {I18n.t(`filters.rules.${types.localeKeysByRule[rule]}`)}
+              </option>
+            ))}
+          </sc.RuleSelect>
+        );
+    }
+  }
+  function renderValueField() {
+    const shouldNotShowValueField =
+      selectedRule === types.rules.IS_BLANK ||
+      selectedRule === types.rules.IS_NOT_BLANK ||
+      selectedRule === types.rules.IS_EMPTY ||
+      selectedRule === types.rules.IS_NOT_EMPTY;
+
+    if (shouldNotShowValueField) return null;
+
+    if (renderCustomValueField) return renderCustomValueField();
+
+    switch (selectedColumnType) {
+      case types.columnTypes.BOOLEAN:
+        return (
+          <sc.ValueInput as={Select} value={`${value}`} onChange={handleChangeBooleanFilterValue}>
```

Comment:
```suggestion
          <sc.ValueInput as={Select} value={value} onChange={handleChangeBooleanFilterValue}>
```

Reply:
This is for a reason, had to convert boolean value to string, I guess use `toString` will be better